standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/QueryWrapper.java

```diff
@@ -0,0 +1,430 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.hive.metastore;
+
+import javax.jdo.Extent;
+import javax.jdo.FetchPlan;
+import javax.jdo.PersistenceManager;
+import javax.jdo.Query;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * A wrapper around the Query object to the caller and let the caller release
```

Comment:
Why we need this class?

If I understand correctly `Query` implements `AutoCloseable`. Would it be enough to use that?

Reply:
Thank you for the review!
The method `close()` of `Query` throws exception,  we should catch the exception explicitly when using use with try-with-resources:
```java
    /**
     * Don't use this method directly; use <code>closeAll()</code> instead. It is intended for use with try-with-resources.
     * @throws Exception if this resource cannot be closed
     */
    void close() throws Exception;
```
The exception would mix with the exception thrown in the `ObjectStore` or `MetaStoreDirectSql`,  make it uneasy to handle on the caller,  as I do not want to break the specified exceptions of the original method.

Reply:
So the `QueryWrapper` is:

_A wrapper around Query objects where the `close()` method is delegated to the wrapped object's `closeAll()` method. This way the users of the wrapper can use try-with-resources without exception handling_

Disclaimer: I am not a native English speaker, so some more refinement might be needed 😄

Reply:
Thank you for your suggestions, that seems nicer.